test/e2e/authorizer/serviceaccounts_test.go

```diff
@@ -0,0 +1,199 @@
+)
+
+func TestServiceAccounts(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	t.Cleanup(cancelFunc)
+
+	server := framework.SharedKcpServer(t)
+	orgClusterName := framework.NewOrganizationFixture(t, server)
+	clusterName := framework.NewWorkspaceFixture(t, server, orgClusterName, "Universal")
+
+	cfg, err := server.DefaultConfig()
+	require.NoError(t, err)
+
+	kubeClusterClient, err := kubernetes.NewClusterForConfig(cfg)
+	require.NoError(t, err)
+
+	kubeClient := kubeClusterClient.Cluster(clusterName)
+
+	t.Log("Creating namespace")
+	namespace, err := kubeClient.CoreV1().Namespaces().Create(ctx, &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "e2e-sa-",
+		},
+	}, metav1.CreateOptions{})
+	require.NoError(t, err, "failed to create namespace")
+
+	t.Log("Creating role to access configmaps")
+	_, err = kubeClient.RbacV1().Roles(namespace.Name).Create(ctx, &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "sa-access-configmap",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"configmaps"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+		},
+	}, metav1.CreateOptions{})
+	require.NoError(t, err, "failed to create role")
+
+	t.Log("Creating role binding to access configmaps")
+	_, err = kubeClient.RbacV1().RoleBindings(namespace.Name).Create(ctx, &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "sa-access-configmap",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      "default",
+				Namespace: namespace.Name,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "Role",
+			Name:     "sa-access-configmap",
+			APIGroup: rbacv1.GroupName,
+		},
+	}, metav1.CreateOptions{})
+	require.NoError(t, err, "failed to create role")
+
+	t.Log("Waiting for service account to be created")
+	require.Eventually(t, func() bool {
+		_, err := kubeClient.CoreV1().ServiceAccounts(namespace.Name).Get(ctx, "default", metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false
+		} else if err != nil {
+			t.Fatalf("unexpected error retrieving service account: %v", err)
+		}
+		return true
+	}, wait.ForeverTestTimeout, time.Millisecond*100, "\"default\" service account not created in namespace %s",
+		helper.QualifiedObjectName(namespace),
+	)
+
+	t.Log("Waiting for service account secret to be created")
+	var tokenSecret corev1.Secret
+	require.Eventually(t, func() bool {
+		secrets, err := kubeClient.CoreV1().Secrets(namespace.Name).List(ctx, metav1.ListOptions{})
+		require.NoError(t, err, "failed to list secrets")
+
+		for _, secret := range secrets.Items {
+			if secret.Annotations[corev1.ServiceAccountNameKey] == "default" {
+				tokenSecret = secret
+				return true
+			}
+		}
+		return false
+	}, wait.ForeverTestTimeout, time.Millisecond*100, "token secret for default service account not created")
+
+	type TokenTestCase struct {
```

Comment:
(No action required) Maybe prefer idiomatic golang test case definition for consistency with existing code? e.g.

```golang
testCases := []struct{
  ...
}{
  ...
}
for _, tc := range testCases {
  ...
}
```

For extra credit, use map[string]struct{} instead of []struct{} to define e2e test cases. The use of a map will randomize execution order which avoids the potential for dependency between tests creeping in.

Reply:
done.